### DISCUSSION TITLE
How can I detect the highlighted text and replace it when the user enters a value into the input field?

### DISCUSSION THREAD

**Discussion by safa-kajabi** [facebook/lexical#5597] [Q&A]
I want to create a modal similar to the one in Slack. If the user highlights text and clicks the link icon, I want to place that text in the text field and prompt the user to enter a link.

<img width="602" alt="Screenshot 2024-02-07 at 22 34 20" src="https://github.com/facebook/lexical/assets/127859730/a2c02cb3-3601-47d9-944f-045a3ac16208">

**Comment by eliottvincent** [comment]
Hey :)

```
let _selectedText, _selection = $getSelection();

if (_selection) {
  _selectedText = _selection.getTextContent();
}
```

Then you can use `_selectedText` and fill your input with that.
